src/discord.coffee

```diff
@@ -39,9 +56,29 @@ class DiscordBot extends Adapter
         @client.on 'ready', @.ready
         @client.on 'message', @.message
         @client.on 'disconnected', @.disconnected
+        @client.on 'messageReactionAdd', (message, user)  => 
+          @.message_reaction('reaction_added', message, user)
+        @client.on 'messageReactionRemove', (message, user) => 
+          @.message_reaction('reaction_removed', message, user)
 
         @client.login(@options.token).catch(@robot.logger.error)
 
+     _map_user: (discord_user, channel_id) -> 
+        user                      = @robot.brain.userForId discord_user.id
+        user.room                 = channel_id
+        user.name                 = discord_user.username
+        user.discriminator        = discord_user.discriminator
+        user.id                   = discord_user.id
+        
```

Comment:
include `discord_user.tag` (it's the full mention text short of the `@`.
If we include `discord_user.toString()`  as well, we can check for mentions.